loader.js

```diff
@@ -34,12 +35,25 @@ module.exports = async function mdxEnhancedLoader(src) {
 
   // Checks if there's a layout, if there is, resolve the layout and wrap the content in it.
   processLayout
-    .call(this, options, data, content, resourcePath)
+    .call(this, options, data, content, resourcePath, scans)
     .then(result => callback(null, result))
     .catch(err => callback(err))
 }
-
-function processLayout(options, frontMatter, content, resourcePath) {
+function scanContent(options, content) {
+  const { mdxEnhancedPluginOptions: pluginOpts } = options
+  if (typeof pluginOpts.scan === 'undefined') return []
+  return pluginOpts.scan
+    .filter(opt => content.match(opt.pattern))
+    .map(opt =>
+      Object.assign({}, opt.passToLayout, {
+        value:
+          typeof opt.passToLayout.value === 'function'
+            ? opt.passToLayout.value(content.match(opt.pattern))
+            : opt.passToLayout.value
+      })
```

Comment:
I feel like we could potentially simplify this by passing back `{ [passToLayout.argumentName]: value }`